### PR TITLE
Update docs

### LINKED ISSUE #402: update documention
super out of date, im going to try to update it before the ohbm educational session

### PR DESCRIPTION
**Pull request by adelavega**
Closes #402 

I updated the docs to reflect new features added in the last few releases. I didn't make any other substantive changes to the docs, a maintenance update. 

In a previous commit (already in master), I added GH Actions to build docs & upload as an artifact for PRs, and upload to `gh-pages` when merged to master.

### CODE REVIEW

**Comment by adelavega** [comment]
I also added a gh-action to build and push docs to the gh-pages branch. I have no clue if it'll work until I merge it, so it might take a few merges to test out.

**Comment by adelavega** [comment]
@rbroc do you mind taking a look at the docs to see if theres anything you want to add or improve? just wanted to get the ball rolling.

@tyarkoni can you take a quick peek at the gh-action to see if it looks like it'll build the docs? There's no documentation on how to run sphinx on this repo so im just guessing.

**Comment by rbroc** [comment]
had a look and pushed a couple of fixes to typos/issues with docstrings.
I think we may still be missing:

- `AudioResamplingFilter` in the Filters section.
- All the misc-type stuff (`SeriesStim`, `MetricExtractor`, `ExtractorResultToSeriesConverter`).
The converter could deserve a little paragraph of its own (either in the converters section, or where Result objects are discussed). Not sure what `_replacements.rst` does, but a few extractors are missing there too.

Re: tutorials and quickstart (though not relevant to this PR), probably worth developing a section for Bert, AudioSet and other DL models where complexity scales a bit. Probably also an example with `MetricExtractor` would be beneficial.

Let's touch base on these points later, happy to help with fixes to docs and/or tutorials. :)

**Comment by adelavega** [comment]
Yes, I think that's what's missing! Feel free to take a stab if you have some free cycles.

**Comment by adelavega** [comment]
@rbroc have you had any time to work on the changes you wanted to make?

**Comment by rbroc** [comment]
done now - maybe do take a quick look just to make sure I did it right ;)

File: docs/_includes/_replacements.rst [hunk 1]
```diff
@@ -1,30 +1,106 @@
...
+
 
 .. currentmodule:: pliers.filters
 .. |Filter| replace:: :py:class:`Filter`
 .. |FrameSamplingFilter| replace:: :py:class:`FrameSamplingFilter`
 .. |ImageCroppingFilter| replace:: :py:class:`ImageCroppingFilter`
+.. |PillowImageFilter| replace:: :py:class:`PillowImageFilter`
+.. |PunctuationRemovalFilter| replace :py:class:`PunctuationRemovalFilter`
+.. |TokenizingFilter| replace :py:class:`TokenizingFilter`
+.. |TokenRemovalFilter| replace :py:class:`TokenRemovalFilter`
```
Comment (adelavega) [inline docs/_includes/_replacements.rst:101]: Why is this happening? Anyone have any idea? Looks like a valid import?

@tyarkoni ?

**Comment by adelavega** [comment]
Roberta, looks good, however now I'm realizing something is going wrong w/ Sphinx. The substitutions are failing at sometimes that I wouldn't expect them to. 

Also, the autodefinition tables are blank. 

Take a look at the artifact: https://github.com/PsychoinformaticsLab/pliers/suites/2267564946/artifacts/47272079

I'm not sure what's going on, but some of this stems from the prior PR it seems, so perhaps its a more general issue with the GH Actions.

Anyone have any clues? @tyarkoni ?

**Comment by adelavega** [comment]
See how the autodoc is yielding empty docstrings: http://psychoinformaticslab.github.io/pliers/reference.html#module-pliers.extractors

**Comment by adelavega** [comment]
Okay, that was a pain but I fixed it.  Sphinx could not find pliers. had to remake the entire action w/ custom code but now we're not relying on an obscure github action that wasn't doing much in the first place.

still a few more minor sphinx errors to be fixed but otherwise, this should be close

File: README.md [hunk 2]
```diff
@@ -1,15 +1,9 @@
-# pliers
```
Comment (jdkent) [inline README.md:1]: If you want to be even more radical (and recycle the README.md in the docs, you can convert the [readme to rest](https://pandoc.org/try/?text=&from=markdown&to=rst&standalone=0) and include the [README.rst in index.rst like fmriprep does](https://github.com/nipreps/fmriprep/blame/4bb8a61cde27575865cdd2b7df5afcb5d6860523/docs/index.rst#L7)

Comment (adelavega) [reply README.md:1]: That's basically what I was too lazy to do but its a good idea

File: docs/index.rst [hunk 3]
```diff
@@ -1,6 +1,81 @@
...
+.. _user guide: http://psychoinformaticslab.github.io/pliers/
+.. _API Reference: http://psychoinformaticslab.github.io/pliers/reference.html
+
+.. |PyPI version fury.io| image:: https://badge.fury.io/py/pliers.svg
+   :target: https://pypi.python.org/pypi/pliers/
+.. |pytest| image:: https://github.com/PsychoinformaticsLab/pliers/actions/workflows/python-package.yml/badge.svg
+.. |Coverage Status| image:: https://coveralls.io/repos/github/tyarkoni/pliers/badge.svg?branch=master
+   :target: https://coveralls.io/github/tyarkoni/pliers?branch=master
+.. |DOI:10.1145/3097983.3098075| image:: https://zenodo.org/badge/DOI/10.1145/3097983.3098075.svg
+   :target: https://doi.org/10.1145/3097983.3098075
```
Comment (jdkent) [inline docs/index.rst:77]:
```suggestion
.. include:: ../README.rst
```
Will it work like how fmriprep does it (so you don't need to copy/paste/maintain the same information)?

Comment (adelavega) [reply docs/index.rst:77]: No but I'm not too worried about it for now as this rarely changes but next time we changed it we can look into that

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+you’re missing a depedency) or you may install all the required
+dependencies:
+
+   pip install -r optional-dependencies.txt
+
+Note, however, that some of these Python dependencies have their own
+(possibly platform-dependent) requirements. For example, python-magic
+requires libmagic and without this, you’ll be relegated to loading all
+your stims explicitly rather than passing in filenames (i.e.,
+``stim = VideoStim('my_video.mp4')`` will work fine, but passi
```
Comment (jdkent) [inline README.rst:64]: it does look like the README is cut off?!?

Comment (tyarkoni) [reply README.rst:64]: Wild. I've always suspected @adelavega doesn't know how to operate copy and paste ;)

Comment (adelavega) [reply README.rst:64]: heh

Comment (jdkent) [reply README.rst]: sounds right, must be a limit on the website, I guess you need to run pandoc locally.

Comment (adelavega) [reply README.rst]: I did now. Now something seems to be failing w/ sphinx grr

Comment (jdkent) [reply README.rst]: that's odd: something to do with this changing?
success: https://github.com/PsychoinformaticsLab/pliers/runs/2169953211?check_suite_focus=true#step:5:15
failure: https://github.com/PsychoinformaticsLab/pliers/runs/2186864097#step:5:14

I wonder why that changed? would manually adding `/home/runner/.local/bin` to PATH fix it?

Comment (adelavega) [reply README.rst]: That did it. Weird.

**Pull request merged**